api/src/main/java/org/apache/cloudstack/api/BaseListDomainResourcesCmd.java

```diff
@@ -42,7 +45,10 @@ public boolean listAll() {
     @Override
     public boolean isRecursive() {
         if (listAll()) {
-            return recursive == null ? true : recursive;
+            Account caller = CallContext.current().getCallingAccount();
+            if (caller.getType() != Account.Type.NORMAL) {
+                return recursive == null ? true : recursive;
+            }
         }
         return recursive == null ? false : recursive;
```

Comment:
Could probably be simplified to:
```suggestion
        return recursive != null && recursive;
```

Reply:
hm, that is not more expressive, i'd say.